app/src/main/java/fr/free/nrw/commons/auth/login/LoginClient.kt

```diff
@@ -106,6 +107,35 @@ class LoginClient(private val loginInterface: LoginInterface) {
         })
     }
 
+    fun doLogin(
+        username: String,
+        password: String,
+        twoFactorCode: String? = null,
+        userLanguage: String = Locale.getDefault().language,
```

Comment:
Are these default parameter values actually needed, and can we drop the import of `Locale`?  Looks like the `LoginActivity` is passing them.  Also, _nullability_ - the twoFactorCode is required to be non-null in the Activity.  You might be able to drop the "?" from it here.

Reply:
Fixed.